src/Farmer/Arm/Storage.fs

```diff
@@ -16,6 +16,7 @@ type StorageAccount =
     { Name : StorageAccountName
       Location : Location
       Sku : Sku
+      DefaultBlobAccessTier: BlobAccessTier Option
```

Comment:
I'm not sure at the ARM level that we should be moving this (probably makes sense at the Builder level though). Does that default access tier apply across all different storage kinds e.g. General Purpose, Blobs, Files etc. or just a subset of them?

Reply:
Looking into this further, while it's valid to set for all kinds except StorageV1, it only seems to affect StorageV2 and Blob Storage. 
Would you prefer adding the accessTier to the storageV2 sku?
Given that the SKU is used at both arm and builder levels, I'm not sure how best to separate those?

Reply:
Mmmm. Either we'll need a new type just for the builder (seems overkill) or we put that default flag against just those two kinds at both builder + ARM levels. We can still expose it as a keyword at the builder level - if someone tries to set it against an invalid storage account kind, throw an exception.